parser/lexer/state.go

```diff
@@ -221,23 +221,14 @@ func numberState(l *lexer) stateFn {
 		switch r {
 		case 'b':
 			l.scanBinaryRemainder()
-			if l.endOffset-l.startOffset <= 2 {
```

Comment:
The parser also reported these issues. Remove from the lexer and improve the parser errors